tests/rptest/tests/rbac_test.py

```diff
@@ -142,3 +161,30 @@ def test_invalid_create_role(self):
 
             with expect_http_error(400):
                 self.superuser_admin.create_role(role=invalid_rolename)
+
+    @cluster(num_nodes=3)
+    def test_list_role_filter(self):
+        role_one = "aaaa-1"
+        role_two = "aaaa-2"
+        role_three = "bbbb-3"
+
+        self._create_and_wait_for_role(role=role_one)
+        self._create_and_wait_for_role(role=role_two)
+        self._create_and_wait_for_role(role=role_three)
+
+        def expect_roles_with_filter(expected_roles: List[str],
+                                     filter: str | None = None,
+                                     principal: str | None = None):
+            res = self.superuser_admin.list_roles(filter=filter,
+                                                  principal=principal)
+            roles = RolesList.from_response(res).roles
+            role_names = map(lambda role: role.name, roles)
+            assert set(expected_roles) == set(role_names)
```

Comment:
Added the message to the assertion.

I tend to write 2 kinds of helpers in my tests, utility functions that derive values/manipulate data/etc and test runner functions that exercise a test pattern that I can call with many different inputs. I think both are useful to reduce repetition, they just have different intentions. I think it makes sense to have assertions in the latter, and I think of this function as the latter. That's also why I have it inside the test rather than as a helper in the class.

I think this should be fine for the stack trace, we should just get multiple stacks inside `rbac_test.py` if it fails. I'm not sure what you mean by the _more descriptive assert logs_.